src/Test/L0/Worker/StepsRunnerL0.cs

```diff
@@ -508,8 +544,11 @@ private Mock<IActionRunner> CreateStep(TaskResult result, string condition, Bool
                         stepContext.Object.Result = r;
                     }
                 });
+            var trace = hc.GetTrace();
+            stepContext.Setup(x => x.Write(It.IsAny<string>(), It.IsAny<string>())).Callback((string tag, string message) => { trace.Info($"[{tag}]{message}"); });
             stepContext.Object.Result = result;
             step.Setup(x => x.ExecutionContext).Returns(stepContext.Object);
+            step.Setup(x => x.RunAsync()).Callback(() => { _stepContext.SetOutput(null, name, "test", "something", out string reference); }).Returns(Task.CompletedTask);
```

Comment:
Consider doing this in the test that needs it (PopulateEnvContextAfterSetupStepsContext), not the shared test setup code

Reply:
will do. :)